### PR TITLE
add option to enable javascript in android webview

### PR DESCRIPTION
**Pull request by rydein**
Android webview disable javascript by default, and added new option to enable javascirpt.

### CODE REVIEW

**Comment by rydein** [comment]
I signed CLA!

**Comment by mehmetf** [comment]
Fixes https://github.com/flutter/flutter/issues/15316

File: packages/url_launcher/example/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -298,7 +298,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
+			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n";
```
Comment (mehmetf) [inline packages/url_launcher/example/ios/Runner.xcodeproj/project.pbxproj]: Please revert changes to this file.

File: packages/url_launcher/example/lib/main.dart [hunk 2]
```diff
@@ -51,6 +51,15 @@ class _MyHomePageState extends State<MyHomePage> {
     }
   }
 
+  Future<Null> _launchInWebViewWithJavaScript(String url) async {
+    if (await canLaunch(url)) {
+      await launch(url,
+          forceSafariVC: true, forceWebView: true, enableJavaScript: true);
```
Comment (mehmetf) [inline packages/url_launcher/example/lib/main.dart]: Nit: Either put all args on one line or use trailing comma (put a comma after the last argument).

File: packages/url_launcher/lib/url_launcher.dart [hunk 3]
```diff
@@ -30,6 +30,9 @@ const MethodChannel _channel = MethodChannel('plugins.flutter.io/url_launcher');
 /// Note that if any of the above are set to true but the URL is not a web URL,
 /// this will throw a [PlatformException].
 ///
+/// [enableJavaScript] is an Android only setting. If true, and [forceWebView]
```
Comment (mehmetf) [inline packages/url_launcher/lib/url_launcher.dart]: I would put this description together with `forceWebView` description in the same paragraph.

File: packages/url_launcher/lib/url_launcher.dart [hunk 4]
```diff
@@ -39,7 +41,8 @@ Future<void> launch(
   String urlString, {
   bool forceSafariVC,
   bool forceWebView,
-  Brightness statusBarBrightness,
+  bool enableJavaScript,
+  Brightness statusBarBrightness = Brightness.light,
```
Comment (mehmetf) [inline packages/url_launcher/lib/url_launcher.dart]: I don't think this is merged correctly. `statusBarBrightness` needs to default to null.

**Comment by alexlovar** [comment]
hi, how to use the depencie if the version 4.0.1 is unpublished. i need to use the javascript support.
i added to yaml file this:
  url_launcher:
   git:
         url: git://github.com/flutter/plugins/blob/master/packages/url_launcher.git
but this dowes not work.

thanks for your help!

**Comment by mehmetf** [comment]
Pub has been updated.

**Pull request merged**